Makefile

```diff
@@ -267,7 +267,10 @@ rpm:
 
 	# Copy spec file to rpmbuild specs directory
 	mkdir -p $(RPMBUILDDIR)/SPECS
-	sed -e 's/@version@/$(PACKAGE_VERSION)/g; s/@release@/$(PACKAGE_RELEASE)/g' contrib/centos/aziot-identity-service.spec >$(RPMBUILDDIR)/SPECS/aziot-identity-service.spec
+	ENGINESDIR='/usr/lib64/openssl/engines'; \
+	ENGINESDIR=$$(if [ ! -d $$ENGINESDIR ]; then echo $$(openssl version -e | sed 's/^ENGINESDIR: "\(.*\)"$$/\1/'); else echo $$ENGINESDIR; fi); \
```

Comment:
Running `openssl` should have higher preference than checking the directory, as in the snippet I'd given, because ultimately what `openssl` says is the enginesdir is what matters.

Reply:
`openssl` on CentOS 7 (v1.02) doesn't have a `version -e` option to respond with the enginesdir.

Reply:
Of course. That's why we had to hard-code the directory in the first place, and why the snippet I gave specifically accounts for that by checking if the output is empty.

Reply:
In your snippet was the `which openssl` needed for something?

Reply:
Yes. The comment on the same line explains its purpose.